var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -150,4 +150,4 @@ def link_tau_arch_dirs(self):
     def setup_environment(self, spack_env, run_env):
         pattern = join_path(self.prefix.lib, 'Makefile.*')
         files = glob.glob(pattern)
-        run_env.set('TAU_MAKEFILE', files[0])
+        run_env.set('TAU_MAKEFILE', files[0] if files else '')
```

Comment:
@pramodk I think you should add a comment explaining why you do this conditional. Something like:
```python
# This function is called both at install time to set up
# the build environment and after install to generate the associated
# module file. In the former case there is no `self.prefix.lib` directory
# to inspect. The conditional below will set `TAU_MAKEFILE` in the
# latter case.
if files:
    run_env.set('TAU_MAKEFILE', files[0])
```
or if you prefer to be concise:
```python
try:
    run_env.set('TAU_MAKEFILE', files[0])
except IndexError:
    # Before the installation takes place
    # the glob expression won't find anything
    pass
```
It may be evident now, but it won't be in a month or two.